optex/base/more-macros.opm

```diff
@@ -194,42 +194,17 @@
 \_public \scantoeol ;
 
    \_doc -----------------------------
-   The \`\replstring``\macro{<textA>}{<textB>}`
+   The \`\replstring``\macro [nested]{<textA>}{<textB>}`
    replaces all occurrences of `<textA>` by `<textB>` in the `\macro` body.
-   The `\macro` must be defined without parameters. The occurrences of
+   The `\macro` must be defined without parameters. Unless the optional
+   `nested` keyword appears after `\macro`, the occurrences of
    `<textA>` are not replaced if they are \"hidden" in braces, for example
    `...{...<textA>...}...`. The category codes in the `<textA>` must exactly
-   match.
-
-   How it works: `\replstring\foo{<textA>}{<textB>}` prepares
-   `\_replacestringsA#1<textA>{...}` and runs
-   `\_replacestringsA<foo-body>?<textA>!<textA>`.
-   So, `#1` includes the first part of <foo-body> before first <textA>.
-   It is saved to \`\_tmptoks` and `\_replacestringsB` is run in a loop.
-   It finishes processing or appends the next part to `\_tmptoks` separated by
-   <textB> and continues loop. The final part of the macro removes the last `?`
-   from resulting `\_tmptoks` and defines a new version of the `\foo`.
-
-   The \^`\replstring` macro is not expandable, but you can create your
-   expandable macro, for example:
-   \begtt
-   \def\replAB#1{\immediateassigned{\def\tmp{#1}\replstring\tmp{A}{B}}\the\_tmptoks}
-   \replAB {text A \and A} % expands to "text B \and B"
-   \endtt
+   match. Only if the `nested` keyword appears after `\macro`, barces can appear in
+   `<textA>`. The \^`\replstring` macro is expandable. 
    \_cod -----------------------------
 
-\_newtoks\_tmptoks
-\_catcode`!=3 \_catcode`?=3
-\_def\_replstring #1#2#3{%  \replstring #1{stringA}{stringB}
-   \_long\_def\_replacestringsA##1#2{\_tmptoks{##1}\_replacestringsB}%
-   \_long\_def\_replacestringsB##1#2{\_ifx!##1\_relax \_else \_toksapp\_tmptoks{#3##1}%
-                                     \_ea\_replacestringsB\_fi}%
-   \_ea\_replacestringsA #1?#2!#2%
-   \_long\_def\_replacestringsA##1?{\_tmptoks{##1}\_edef#1{\_the\_tmptoks}}%
-   \_ea\_replacestringsA \_the\_tmptoks}
-\_normalcatcodes
-
-\_public \replstring ;
+\_def\replstring{\_replstring}
```

Comment:
@olsak I wasn't sure if using `\_let` would be better here, because then the order of the files would be significant (and I'm not sure how lua code interacts at format generation time as well), so I just went with `\_def` as I saw it was done with `\_addpageresource`.